test/e2e_node/pod_resize_test.go

```
@@ -0,0 +1,1588 @@
+				cpuQuota = -1
+			}
+			cpuLimitString = strconv.FormatInt(cpuQuota, 10)
+			if podOnCgroupv2Node {
+				if cpuLimitString == "-1" {
+					cpuLimitString = "max"
+				}
+				cpuLimitString = fmt.Sprintf("%s %s", cpuLimitString, CPUPeriod)
+			}
+			memLimitString = strconv.FormatInt(memLimitInBytes, 10)
+			if podOnCgroupv2Node && memLimitString == "0" {
+				memLimitString = "max"
+			}
+			if memLimitString != "0" {
+				if !verifyCgroupValue(ci.Name, cgroupMemLimit, memLimitString) {
+					return false
+				}
+			}
+			if !verifyCgroupValue(ci.Name, cgroupCPULimit, cpuLimitString) {
+				return false
+			}
+			if podOnCgroupv2Node {
+				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+				cpuShares = int64(1 + ((cpuShares-2)*9999)/262142)
+			}
+			if !verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(cpuShares, 10)) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func waitForPodResizeActuation(ctx context.Context, f *framework.Framework, c clientset.Interface, podClient *e2epod.PodClient, pod, patchedPod *v1.Pod, expectedContainers []TestContainerInfo) *v1.Pod {
+
+	waitForContainerRestart := func() error {
+		var restartContainersExpected []string
+		for _, ci := range expectedContainers {
+			if ci.RestartCount > 0 {
+				restartContainersExpected = append(restartContainersExpected, ci.Name)
+			}
+		}
+		if len(restartContainersExpected) == 0 {
+			return nil
+		}
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			pod, err := podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			restartedContainersCount := 0
+			for _, cName := range restartContainersExpected {
+				cs, _ := podutil.GetContainerStatus(pod.Status.ContainerStatuses, cName)
+				if cs.RestartCount < 1 {
+					break
+				}
+				restartedContainersCount++
+			}
+			if restartedContainersCount == len(restartContainersExpected) {
+				return nil
+			}
```

This looks brittle to me. At least fail fast if `restartContainersCount > restartContainersExpected`

---

Done and tested, works fine proposal but not sure why it looked brittle, can you please elaborate.